From 364956064eab5e53c7eb4af4ec78888784cfc9d2 Mon Sep 17 00:00:00 2001
From: Patrick Uiterwijk <puiterwijk@redhat.com>
Date: Tue, 13 Dec 2016 13:48:58 +0000
Subject: [PATCH 3/4] Make KojiWrapper parse krb_rdns

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>
---
 pungi/wrappers/kojiwrapper.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pungi/wrappers/kojiwrapper.py b/pungi/wrappers/kojiwrapper.py
index 85a933a..8dcb5aa 100644
--- a/pungi/wrappers/kojiwrapper.py
+++ b/pungi/wrappers/kojiwrapper.py
@@ -39,7 +39,7 @@ class KojiWrapper(object):
             for key in ('krbservice', 'timeout', 'keepalive',
                         'max_retries', 'retry_interval', 'anon_retry',
                         'offline_retry', 'offline_retry_interval',
-                        'debug', 'debug_xmlrpc',
+                        'debug', 'debug_xmlrpc', 'krb_rdns',
                         'use_fast_upload'):
                 value = getattr(self.koji_module.config, key, None)
                 if value is not None:
-- 
2.9.3

